api/src/main/java/org/openmrs/api/context/Daemon.java

```diff
@@ -184,6 +188,40 @@ public static void executeScheduledTask(final Task task) throws Exception {
 			}
 		}
 	}
+
+	/**
+	 * Authenticates a daemon user with privileges so the task can call service layers <br>
+	 * <br>
+	 * This can only be called from {@link TimerSchedulerTask} during actual task execution
+	 *
+	 * <strong>Should</strong> not be called from other methods other than TimerSchedulerTask
+	 * <strong>Should</strong> not throw error if called from a TimerSchedulerTask class
+	 */
+	public static void authenticateTask() throws Exception {
```

Comment:
@dkayiwa This Daemon class is very layered, can I get more info on how it operates